### PR TITLE
Xbox Controls Finished Mapping and Working, Auto Climb Routines, Updated Intake Functionality

### PR DESCRIPTION
**Pull request by El1zabethS**
**_Added:_**

Commands: 
- Intake Go To - Calls Right and Left Intake Go To
- Right Intake Go To - Tells the Right Actuator Arm to go to a Specific Encoder Value
- Left Intake Go To - Tells the Left Actuator Arm to go to a Specific Encoder Value
- Climber Go To - Tells the Climber to Go to a Specific Encoder Value
- Tilt To - Moves the intake Actuators until the Robot is Tilted at a Specific Angle
- End Game Climb Mid - Uses Intake and Climber Go To to get hooked onto Mid Bar with just the Intake
- End Game Climb High- Uses Intake Go To, Climber Go To, and Tilt To to get hooked onto High Bar with just the Intake ( Can also be used for Traverse Climb ending with just Intake)
- End Game Climb Traverse - Uses Intake Go To, Climber Go To, and Tilt To to get hooked onto High Bar with just the Climber (AKA Alt Traverse)
- Climber Reset - Resets Climber Encoder Value at the Bottom

Xbox Controller:
- Switched Joysticks for Xbox Controllers so now Aux and Driver use Controllers
- We now can use the triggers on the Xbox Controllers
- _For Driver -_
- The Right Bumper sends the Manually Climber Up, The Left Bumper sends the Manually Climber Down
- Left Trigger runs the Intake Backwards, Right Trigger runs the Intake Forwards
- Left Joystick Moves Forward and Back, Right Joystick Moves Left and Right
- Start (the weird button in the middle right) calls Climber Reset, Back (The other weird Button) Initiates Climb High Bar Routine/ Main Traversal Climb
- _For Aux -_ 
- Same Controls as Before except for Added Mid Bar Climb to Right Trigger and Alt Traverse Climb to Left Trigger

**_Deleted:_**

The command "Regurgitate" Was removed because it was asked to be added for regionals, however, was not used and is pointless. Plus, nobody really knew/remembered that button

**_Changed:_**

Constants Folder is now more Organized

Intake:
- Now can read the right encoder value.
- Rather than reading the left actuator and telling the right actuator to do the same, it individually reads both and feeds them the same distances (Basically switched from following to checking both sides.)
- When using "Intake Go To" it will automatically fix the arms and put them at the same distance
- Commands were changed as seen fit to deal with these new changes (Such as Actuator Down now instantiates both arms rather than just one)

Other:
- Smaller things may have been changed, but it would be for the above commands and either shouldn't change or will add functionality. (Example: Added Reading Roll to Gyro Sub for Tilt To)

### CODE REVIEW

**Comment by El1zabethS** [comment]
Note:  
- The Climbing still isn't perfect
- I don't know how to squash commits (sorry Mr. Bruce :( )

**Comment by INeedCaffeine** [comment]
LOTS of different changes in this PR.  It really should have been at least 3 separate PRs.   Please avoid mega PRs in the future.

**Comment by INeedCaffeine** [comment]
When processing the Merge Request I think there is a checkbox below the merge button for squashing commits.  

It may now be an option on the button itself if there is a pulldown option on it (see https://cloudfour.com/wp-content/uploads/2019/10/github-merge-options.png )

You can read more about it here: https://cloudfour.com/thinks/squashing-your-pull-requests/

File: src/main/java/frc/robot/commands/EndGameClimbTraverse.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    addCommands(
+      new TiltTo(-40),
+      new ClimberGoTo(48), 
+      new WaitCommand(2),
+      new TiltTo(-30),
+      new WaitCommand(2),
+      new ClimberGoTo(40),
+      new ClimberGoTo(25),
+      parallel(new ClimberGoTo(20), new intakeGoTo(-15)),
+      new intakeGoTo(-4),
```
Comment (J-Palomino) [inline src/main/java/frc/robot/commands/EndGameClimbTraverse.java:28]: Does this need to change to -2?

File: src/main/java/frc/robot/commands/IntakeDown.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// Copyright (c) FIRST and other WPILib contributors.
```
Comment (J-Palomino) [inline src/main/java/frc/robot/commands/IntakeDown.java:1]: REMOVE IF NOT USING

File: src/main/java/frc/robot/commands/TiltTo.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+RobotContainer.m_intakeSubsystem.stopActuators();
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+
+  if(RobotContainer.m_gyroSubsystem.getRoll() > Angle){
+    RobotContainer.m_intakeSubsystem.setRightActuatorDown(0.2);
+    RobotContainer.m_intakeSubsystem.setLeftActuatorDown(0.2);
```
Comment (J-Palomino) [inline src/main/java/frc/robot/commands/TiltTo.java:33]: TEST THIS!!!!

File: src/main/java/frc/robot/commands/leftIntakeGoTo.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    } 
+  else {
+      RobotContainer.m_intakeSubsystem.stopLeftActuator();
+    }
+  }
+
+  // Called once the command ends or is interrupted.
+  @Override
+  public void end(boolean interrupted) {
+    RobotContainer.m_intakeSubsystem.stopActuators();
```
Comment (J-Palomino) [inline src/main/java/frc/robot/commands/leftIntakeGoTo.java:48]: TEST THIS to make sure the left arm isnt stopping the right arm and vice versa

File: src/main/java/frc/robot/commands/rightIntakeGoTo.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  public rightIntakeGoTo(double Endpoint) {
+    // Use addRequirements() here to declare subsystem dependencies.
+    endpoint = -Endpoint;
+
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    RobotContainer.m_intakeSubsystem.stopActuators();
```
Comment (J-Palomino) [inline src/main/java/frc/robot/commands/rightIntakeGoTo.java:26]: TEST this to  make surte the right intaker arm isnt stopping the left intake arm and vice versa

File: src/main/java/frc/robot/subsystems/ClimberSubsystem.java [hunk 6]
```diff
@@ -19,9 +21,10 @@ public class ClimberSubsystem extends SubsystemBase {
   public ClimberSubsystem() {
     
     TalonFXConfiguration config = new TalonFXConfiguration();
-    config.statorCurrLimit.enable = false;
-    config.statorCurrLimit.currentLimit = 30;
+    config.statorCurrLimit.enable = true;
+    config.statorCurrLimit.currentLimit = 50;
     climberMotor.configAllSettings(config); 
+    Position = climberMotor.getSelectedSensorPosition()/6380;
```
Comment (J-Palomino) [inline src/main/java/frc/robot/subsystems/ClimberSubsystem.java:27]: PLEASE put 6380 into a constant

File: src/main/java/frc/robot/subsystems/ClimberSubsystem.java [hunk 7]
```diff
@@ -68,4 +73,30 @@ public void climberPosition() {
   public void periodic() {
     // This method will be called once per scheduler run
   }
+
+  public double getCurrentPosition(){
+    return SmartDashboard.getNumber("Climber Encoder", 0);
```
Comment (J-Palomino) [inline src/main/java/frc/robot/subsystems/ClimberSubsystem.java:78]: update to get directly from the Gyro

File: src/main/java/frc/robot/subsystems/IntakeSubsystem.java [hunk 8]
```diff
@@ -64,47 +68,116 @@ public void stopMotor() {
...
     } else {
       // We are going down but bottom limit is not tripped so go at commanded speed
       intakeLeftActuator.set(-speed);
-      intakeRightActuator.set(speed);
       down = false;
     }
     armPosition();
   }
+
+ public void climbingCurrentLimit(){
```
Comment (J-Palomino) [inline src/main/java/frc/robot/subsystems/IntakeSubsystem.java:170]: Double check this doesnt need cleanup

**Comment by INeedCaffeine** [comment]
This PR is now too stale to merge in.  

Many of these changes have been picked up indirectly by merging in other branches because we did not use branches the way we should have.  

Some of the review comments may still be valid but at this point in the season all may not be worth addressing.  The cleanup ones are definitely worth it to avoid code cruft.

**Pull request closed without merging**